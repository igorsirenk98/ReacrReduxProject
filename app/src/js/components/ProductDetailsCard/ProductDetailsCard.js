import React from 'react';

import Typography from '@material-ui/core/Typography';
import Card from '@material-ui/core/Card';
import CardContent from '@material-ui/core/CardContent';

const ProductDetailsCard = (props) => {
	const { product } = props;

	return (
		<>
			{!!Object.entries(product).length
				&& (
					<Card className="card">
						<img
							className="img"
							src={`data:image/jpeg;base64, ${product.largePhoto}`}
							alt={product.name}
						/>
						<CardContent>
							<Typography gutterBottom variant="h5" component="h2">
								{product.name}
							</Typography>
							<Typography>Product Code:</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								{product.productNumber}
							</Typography>
							<Typography>Type:</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								{product['productSubcategory.name']}
							</Typography>
							<Typography>Price:</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								{product.listPrice}$
							</Typography>
							<Typography>Characteristics:</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								Color: {product.color}
							</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								Weight: {product.weight}{product.weightUnitMeasureCode}
							</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								Size: {product.size}{product.sizeUnitMeasureCode}
							</Typography>
							<Typography>Description:</Typography>
							<Typography variant="body2" color="textSecondary" component="p">
								{product.description}
							</Typography>
						</CardContent>
					</Card>
				)}
		</>
	);
};

export default ProductDetailsCard;
