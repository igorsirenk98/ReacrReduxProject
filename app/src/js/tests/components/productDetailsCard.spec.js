import React from 'react'
import { shallow } from 'enzyme'
import ProductDetailsCard from '../../components/ProductDetailsCard/ProductDetailsCard';

describe('ProductDetailsCard component', () => {
    const productDetailsCard = shallow(<ProductDetailsCard product={{
        productId: 1,
        name: 'bike',
        productNumber: 2,
        color: 'red',
        weight: 75,
        weightUnitMeasureCode: 'kg',
        size: 24,
        sizeUnitMeasureCode: 'cm',
        productSubcategory: {
            name: 'road'
        },
        listPrice: 100,
        description: 'Something...'
    }} />);

    it('renders correctly', () => {
        expect(productDetailsCard.debug()).toMatchSnapshot()
    })
})
