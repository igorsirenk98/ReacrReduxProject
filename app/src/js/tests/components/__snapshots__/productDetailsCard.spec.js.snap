// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetailsCard component renders correctly 1`] = `
"<Fragment>
  <WithStyles(ForwardRef(Card)) className=\\"card\\">
    <img className=\\"img\\" src=\\"data:image/jpeg;base64, undefined\\" alt=\\"bike\\" />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
        bike
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))>
        Product Code:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
        2
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))>
        Type:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\" />
      <WithStyles(ForwardRef(Typography))>
        Price:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
        100
        $
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))>
        Characteristics:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
        Color: 
        red
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
        Weight: 
        75
        kg
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
        Size: 
        24
        cm
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))>
        Description:
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" color=\\"textSecondary\\" component=\\"p\\">
        Something...
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(Card))>
</Fragment>"
`;
